Lib/argparse.py

```diff
@@ -2543,9 +2543,18 @@ def _get_value(self, action, arg_string):
     def _check_value(self, action, value):
         # converted value must be one of the choices (if specified)
         if action.choices is not None and value not in action.choices:
-            args = {'value': value,
-                    'choices': ', '.join(map(repr, action.choices))}
-            msg = _('invalid choice: %(value)r (choose from %(choices)s)')
+            closest_choice = _difflib.get_close_matches(value, action.choices)
```

Comment:
this might fail if `action.choices` are not strings

Reply:
argparser not accept other than string, it will throw error even before reaching at this line
reference ==> https://stackoverflow.com/questions/49578928/typeerror-int-object-is-not-subscriptable-when-i-try-to-pass-three-arguments

Reply:
The second example for the documentation (https://docs.python.org/3/library/argparse.html#choices)
```python
parser = argparse.ArgumentParser(prog='doors.py')
parser.add_argument('door', type=int, choices=range(1, 4))
print(parser.parse_args(['3']))
```

Also - I would use 
```suggestion
            closest_choice = _difflib.get_close_matches(value, action.choices, 1)
```